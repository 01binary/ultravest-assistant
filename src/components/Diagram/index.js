import { h } from 'preact';
import Segment from '../Segment';
import style from './style';

/**
 * Burnout diagram.
 * @param {object[]} segments - The program segments.
 * @returns {JSX.Element} - A stateless component. 
 */
const Diagram = ({ segments }) => (
	<section class={style.diagram}>
		{ segments.map((segment, index) => (
			<Segment
				prev={segments[index - 1]}
				cur={segment}
			/>
		))}
	</section>
);

export default Diagram;