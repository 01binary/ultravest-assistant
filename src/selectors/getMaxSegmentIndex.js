/**
 * Get index of a segment with the highest temperature.
 * @param {object[]} segments - The array of segments.
 * @returns {number} - The index of a segment with the highest temperature.
 */
export default segments => segments.reduce(
	(max, segment, index) => (
		segment.temp > max.temp ? { ...segment, index } : max
	), { temp: 0, index: 0 }
).index;
