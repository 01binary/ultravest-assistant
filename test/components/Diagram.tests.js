/*import { h } from 'preact';
import { deep } from 'preact-render-spy';
import Diagram from '../../src/components/Diagram';
import getMaxSegmentIndex from '../../src/selectors/getMaxSegmentIndex';
import presets from '../../src/config/burnoutPresets';

describe('Diagram', () => {

	const first = Object.keys(presets)[0];
	const segments = presets[first].segments;
	const maxIndex = getMaxSegmentIndex(segments);
	const props = { segments, maxIndex };

	let wrapper;

	beforeAll(() => {
		wrapper = deep(
			<Diagram {...props} />
		);
	});

	afterAll(() => {
		wrapper = null;
	});

	test('should render', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
*/