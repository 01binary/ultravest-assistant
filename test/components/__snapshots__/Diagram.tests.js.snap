// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagram should render 1`] = `
preact-render-spy (1 nodes)
-------
<section class="diagram">
  <article class="segment raise">
    <figure class="temperature">
      <svg></svg>
      <figcaption>
        <h3>temp</h3>
        ambient
      </figcaption>
    </figure>
    <figure class="rate">
      <svg></svg>
      <figcaption>
        <h3>rate</h3>
        300°
        <span class="units">F/hr</span>
      </figcaption>
    </figure>
    <figure class="time">
      <svg></svg>
      <figcaption>
        <h3>hold</h3>
        1
        <span class="units">hour</span>
      </figcaption>
    </figure>
  </article>
  <article class="segment raise">
    <figure class="temperature">
      <svg></svg>
      <figcaption>
        <h3>temp</h3>
        700°
        <span class="units">F</span>
      </figcaption>
    </figure>
    <figure class="rate">
      <svg></svg>
      <figcaption>
        <h3>rate</h3>
        400°
        <span class="units">F/hr</span>
      </figcaption>
    </figure>
    <figure class="time">
      <svg></svg>
      <figcaption>
        <h3>hold</h3>
        1
        <span class="units">hour</span>
      </figcaption>
    </figure>
  </article>
  <article class="segment raise">
    <figure class="temperature">
      <svg></svg>
      <figcaption>
        <h3>temp</h3>
        1350°
        <span class="units">F</span>
      </figcaption>
    </figure>
    <figure class="rate">
      <svg></svg>
      <figcaption>
        <h3>rate</h3>
        325°
        <span class="units">F/hr</span>
      </figcaption>
    </figure>
    <figure class="time">
      <svg></svg>
      <figcaption>
        <h3>hold</h3>
        2
        <span class="units">hours</span>
      </figcaption>
    </figure>
  </article>
  <article class="segment lower">
    <figure class="temperature">
      <svg></svg>
      <figcaption>
        <h3>temp</h3>
        700°
        <span class="units">F</span>
      </figcaption>
    </figure>
    <figure class="rate">
      <svg></svg>
      <figcaption>
        <h3>rate</h3>
        700°
        <span class="units">F/hr</span>
      </figcaption>
    </figure>
    <figure class="time">
      <svg></svg>
      <figcaption>
        <h3>hold</h3>
        1
        <span class="units">hour</span>
      </figcaption>
    </figure>
  </article>
</section>

`;
